packages/babel-preset-default/index.js

```diff
@@ -56,6 +56,7 @@ module.exports = function( api ) {
 		presets: [ getPresetEnv() ],
 		plugins: [
 			require.resolve( '@babel/plugin-proposal-object-rest-spread' ),
+			require.resolve( '@wordpress/warning/babel-plugin' ),
```

Comment:
_(**Edit:** I noticed before submitting review that there _is_ a `babel-plugin.js` in the top-level directory. I must have missed it in my first pass. All the same, worth considering the notes about transpilation, especially considering that the top-level directory which would be imported here is referencing _the source_ and not the built file, which means it's pointless for us to include in `src`, and also the built file in `build` is unused)_

I'm a bit confused about how this path actually gets resolved, considering that `babel-plugin` is contained within a `src` directory (or one of the `build(-)` directories after build). For referencing it by this path, I might have expected that it would require that a `babel-plugin.js` file exist at the top-level directory. Is that not the case?

It's also worth pointing out that, unlike other packages where we need to transpile the code using Babel (to be able to run in a browser or other environment with non-modern JS support), I expect that the code of the Babel plugin could be run directly, and not require to be part of the build. There are examples of prior art for this elsewhere, e.g. [`@wordpress/babel-plugin-import-jsx-pragma`](https://github.com/WordPress/gutenberg/tree/master/packages/babel-plugin-import-jsx-pragma), which does not have `src`, and as such is not [built](https://github.com/WordPress/gutenberg/blob/5714888c0acd52fedfc31a64815c25e400f6caac/bin/packages/build.js#L125).

This one is a bit unique in that we have both of: code which doesn't need to be transpiled, and which _does_ need to be transpiled (the `warning` function itself). I don't think there's much harm in just transpiling it all, but I wanted to point it out all the same. And it might be relevant for if we do need a `babel-plugin.js` at the top-level directory (since we could just move it verbatim from where it currently resides).

Reply:
You're right! I moved the `babel-plugin.js` file to the package top-level directory together with its `test` folder, to be consistent with the other babel plugin packages in the repo.